benchmark/buffers/buffer-indexof.js

```diff
@@ -6,26 +6,22 @@ const path = require('path');
 const searchStrings = [
   '@',
   'SQ',
-  '10x',
   '--l',
   'Alice',
   'Gryphon',
-  'Panther',
   'Ou est ma chatte?',
   'found it very',
-  'among mad people',
   'neighbouring pool',
-  'Soo--oop',
   'aaaaaaaaaaaaaaaaa',
   'venture to go near the house till she had brought herself down to',
   '</i> to the Caterpillar',
 ];
 
 const bench = common.createBenchmark(main, {
   search: searchStrings,
-  encoding: ['undefined', 'utf8', 'ucs2', 'binary'],
+  encoding: ['utf8', 'ucs2'],
   type: ['buffer', 'string'],
-  n: [100000]
+  n: [5e4]
```

Comment:
The encoding `undefined` is actually identical to `utf8`. Thus, it's redundant to check both cases because the minimally different execution paths do not make up the actual difference, especially due to the fact that it has nothing to do with `indexof`.

I reduced the search queries by checking the text that is searched for patterns that seem to end in similar parts of the text or that have a similar frequency and length. Multiple seemed redundant and we still check quite a lot even though the performance profile shows almost no difference for either case if they have a similar position and search string length.